### ISSUE TITLE
``pyreverse`` does not handle arguments of form ``Optional[int]`` correctly

### ISSUE THREAD

**Issue by DudeNr33** [PyCQA/pylint#4845]
### Bug description

Given a file ``a.py``:

```python
from typing import Optional


class DoSomething:
    def __init__(self, a_string: str, optional_int: Optional[int] = None):
        self.my_string = a_string
        self.my_int = optional_int
```

### Command used

```shell
pyreverse -ASmy a.py
```

### Pylint output

Output of pyreverse (unfortunately I can't upload images with the new forms?) displays the type of ``my_int`` just as ``Optional``, not as ``Optional[int]``.

Note that if the code is changed to:

```python
class DoSomething:
    def __init__(self, a_string: str, optional_int: int = None):
        self.my_string = a_string
        self.my_int = optional_int

The output correctly displays ``my_int`` as ``Optional[int]``.
```

### Expected behavior

Display ``my_int`` as ``Optional[int]`` in both cases.

### Pylint version

```shell
pylint 2.10.0-dev0
astroid 2.6.6
Python 3.9.6 (default, Jul 28 2021, 21:15:06) 
[Clang 12.0.5 (clang-1205.0.22.9)]
```

### OS / Environment

macOS Big Sur 11.4

### Additional dependencies

I am currently preparing the MR for PlantUML support and extracted the logic to get the type annotations into the ``Printer`` base class. It would probably be best to fix this issue after the MR is merged.

**Comment by mbyrnepr2** [comment]
It seems to work as expected when I reset to commit 1e55ae64624d28c5fe8b63ad7979880ee2e6ef3f and run the provided command. 
Perhaps a recent change affected it @DudeNr33.

**Comment by DudeNr33** [comment]
No that's great, go ahead! Thanks for fixing it!
